src/device/ping-device.cpp

```diff
@@ -0,0 +1,94 @@
+#include <ping-device.h>
+
+#include <ping-message-common.h>
+#include <ping-message.h>
+#include <ping-time.h>
+
+bool PingDevice::initialize() {
+    return request(CommonId::PROTOCOL_VERSION)
+        && request(CommonId::DEVICE_INFORMATION);
+}
+
+ping_message* PingDevice::read() {
+    char b;
+    int result = _port.read(&b, 1);
+    if (result > 0) {
+        if (_parser.parseByte(static_cast<char>(b)) == PingParser::NEW_MESSAGE) {
+        return &_parser.rxMessage;
+        }
+    }
+    return nullptr;
+}
+
+ping_message* PingDevice::request(uint16_t id, int timeoutMs) {
+    common_general_request msg;
```

Comment:
Can we have this variable to be inside the class to avoid the memory allocation every time that this function runs ? 
Can we also use explicit names like: `common_general_request request_message`

Reply:
> Can we have this variable to be inside the class to avoid the memory allocation every time that this function runs ?

yes

> Can we also use explicit names like: common_general_request request_message

I tried it, this complicates the template and reduces readability in the template and generated code IMO.

Reply:
> Can we also use explicit names like: common_general_request request_message

I can change some of the occurrences in ping-device like this one you point out. There is the handlemessage that has the template pains I mentioned.